src/anm/events.js

```diff
@@ -152,26 +152,28 @@ MouseEventsSupport.prototype.adaptEvent = function(event) {
 MouseEventsSupport.prototype.dispatch = function(event) {
     var owner = this.owner;
     var localEvent = this.adaptEvent(event);
-    var dispatchedByChild; // found the matching child inside
+    var dispatchedChild;  // found the matching child inside
     if (owner.inside(localEvent)) {
         owner.reverseEach(function(child) {
             if (child.isActive()) {
-                dispatchedByChild = child.dispatchMouseEvent(localEvent);
-                if (dispatchedByChild) return false; // stop iteration of reverseEach
+                dispatchedChild = child.dispatchMouseEvent(localEvent) ? child : null;
+                if (dispatchedChild) return false; // stop iteration of reverseEach
             }
         });
 
-        if (dispatchedByChild) return true;
-
-        if (event.type === 'mouseclick') {
-            this.owner.fire('mouseclick', localEvent);
-            return true;
-        } else if (event.type === 'mousemove') {
-            this.processMove(localEvent); // fire mouseenter/mousexit if required
-            return true;
+        if (!dispatchedChild && (event.type === 'mousemove')) {
+            this.processMove(localEvent); // fire mouseenter/mouseexit if required
```

Comment:
I think we should fire both at least for the `mouseenter` case—if you listen only for `mousemove`, then you need this first event too. But for `mouseexit` it's controversial, since it will have a `mouseenter` pair in some cases, in some cases it won't.